protocol/http_scanner.go

```diff
@@ -22,7 +22,7 @@ var (
 		"/products", "/products/cat"}
 )
 
-func (hps *HTTPProtocolScanner) ScanProtocol(hiddenService string, proxyAddress string, report *report.OnionScanReport) {
+func (hps *HTTPProtocolScanner) ScanProtocol(hiddenService string, proxyAddress string, directoryDepth int, report *report.OnionScanReport) {
```

Comment:
I'd like to keep this interface uncluttered to allow cross-protocol chaining later on. We probably should include a pointer the the configuration, rather than include every configuration option as a parameter to everything.